x/ccv/provider/migrations/v8/migrations.go

```diff
@@ -211,37 +218,86 @@ func MigrateLaunchedConsumerChains(ctx sdk.Context, store storetypes.KVStore, pk
 		if buf != nil {
 			topN = binary.BigEndian.Uint32(buf)
 		}
+
 		validatorsPowerCapKey := append([]byte{LegacyValidatorsPowerCapKeyPrefix}, []byte(chainId)...)
 		var validatorsPowerCap uint32 = 0
 		buf = store.Get(validatorsPowerCapKey)
 		if buf != nil {
 			validatorsPowerCap = binary.BigEndian.Uint32(buf)
 		}
+
 		validatorSetCapKey := append([]byte{LegacyValidatorSetCapKeyPrefix}, []byte(chainId)...)
 		var validatorSetCap uint32 = 0
 		buf = store.Get(validatorSetCapKey)
 		if buf != nil {
 			validatorSetCap = binary.BigEndian.Uint32(buf)
 		}
+
+		bech32PrefixConsAddr := sdk.GetConfig().GetBech32ConsensusAddrPrefix()
```

Comment:
@mpoke If we are going to set the allowlists and denylists, we might as well add them in the power-shaping parameters. The way I see it, is that a launched chain might execute successfully a `ConsumerModification` and hence allowlists and denylists would be set. If this is the case we want this to show in the `powerShapingParameters` as well.

Reply:
If the allowlist is empty in a `MsgUpdateConsumer`, then the chain won't have an allowlist anymore. It won't have an allowlist in either the `allowlist` state or the state under `PowerShapingParameters.allowlist`.

> So we basically have the same info store twice?
We do. Information is stored in a slightly different ways between `allowlist` and `PowerShapingParameters.allowlist`. We could only use the latter but this would imply we alwyas have to traverse an allowlist when trying to see if a validator can validate a chain.

Reply:
Okay, so we need to make sure the two are always synced -- the fields in the `PowerShapingParameters` and the values under the `allowlist` and `denylist` keys. Right?

Reply:
Yes.